pkg/git/fetch.go

```diff
@@ -91,6 +103,63 @@ func (f Fetcher) Fetch(dir, gitURL, gitRevision, metadataDir string) error {
 	return nil
 }
 
+func copyFile(src, dest string) error {
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer srcFile.Close()
+
+	destFile, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer destFile.Close()
+
+	if _, err = io.Copy(destFile, srcFile); err != nil {
+		return err
+	}
+
+	srcInfo, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	return os.Chmod(dest, srcInfo.Mode())
+}
```

Comment:
noticed duplicate in [cmd/build-init/main.go](https://github.com/pivotal/kpack/blob/de396d3f95f63165aa598d6cd8c23fe31e3e58e6/cmd/build-init/main.go#L223-L246)
wonder if it makes sense to share it in /pkg/* ?

Reply:
I have mixed opinions on this. On one hand I don't like seeing duplicate code, but on the other hand it might be unnecessary to extract one function to a new package if it's only being used in two places.